sdk/lib/actions/input/inputSpecTypes.ts

```diff
@@ -242,13 +216,13 @@ export type ListValueSpecText = {
 }
 export type ListValueSpecObject = {
   type: "object"
-  /** this is a mapped type of the config object at this level, replacing the object's values with specs on those values */
   spec: InputSpec
-  /** indicates whether duplicates can be permitted in the list */
   uniqueBy: UniqueBy
-  /** this should be a handlebars template which can make use of the entire config which corresponds to 'spec' */
   displayAs: string | null
 }
+// TODO Aiden do we really want this expressivity? Why not the below. Also what's with the "readonly" portion?
```

Comment:
There are times that we are reading in a Readonly<Something> that this unique by needed the readonly section. I believe that most of this is because it comes from my Ts-Matches, I make the type the most restrictive which is RO of the array.